pyvista/core/filters/data_set.py

```diff
@@ -2090,7 +2090,7 @@ def glyph(
                 alg.SetIndexModeToOff()
 
         if isinstance(scale, str):
-            dataset.set_active_scalars(scale, preference='cell')
+            dataset.set_active_scalars(scale)
```

Comment:
I'm curious why this may want to be "cell"-preferenced...

Reply:
Threshold is a cell-wise operation.  It does allow operating on point data, but it is still cell-wise thresholding, e.g. it keeps the entire cell with all of its points if one point meets the threshold.  My guess is that this filter is probably more common to use for cell centered values, but this could easily be a biased view since I typically work with cell data.

This will be a breaking change.  I know that this will cause problems for anyone using OpenFOAM data as the reader by default makes point and cell data with the same name.  If we change this, we need a strong warning in the documentation about it.

Reply:
This line/diff is for the `glyph()` filter though. Did you mean to comment above on the change the the `threshold()` method signature? 

> This will be a breaking change. .. will cause problems for anyone using OpenFOAM data as the reader by default makes point and cell data with the same name

This is a good point... (for the threshold and any any other filters). 

I would like to change these to be more consistent though. I'm planning to rework the `threshold()` filter in particular as it's currently yielding different results than paraview and the API upstream has changed a bit. ref #3610

Reply:
With this in mind, I'm going to revert the changes to the `threshold()` operations for now. I think the remaining changes are all low-impact though. I will try to re-work the threshold filter to have the same results as paraview and will save this `preference` change for then

The question still remains why this line in the `glyph()` filter needs to prefer CELL data though

Reply:
> This line/diff is for the `glyph()` filter though.

Oops sloppy reading of the diff.  I don't even think glyph filter works with cell data?

Reply:
I think the glyph filter should accept a preference argument and the default should be `'point'`.  This would be a breaking change, but makes it more consistent and also the glyph filter only operates on point data.  If cell data is selected, then this filter uses `cell_centers` to convert to point data.